src/lib/directives/navigation-mode.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, Input, OnChanges, SimpleChanges } from '@angular/core';
+
+import { NavigationMode } from '../navigation/navigation-mode.interface';
+import { ConfigurableNavigationMode } from '../navigation/configurable-navigation-mode';
+import { WizardComponent } from '../components/wizard.component';
+
+
+@Directive({
+  selector: '[awNavigationMode]'
+})
+export class NavigationModeDirective implements OnChanges {
+
+  @Input()
+  public awNavigationMode: NavigationMode|null;
+
+  @Input()
+  public navigateBackward: 'allow'|'deny'|null;
+
+  @Input()
+  public navigateForward: 'allow'|'deny'|null;
+
+  constructor(private wizard: WizardComponent) { }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.wizard.updateNavigationMode(this.getNavigationMode());
+  }
+
+  private getNavigationMode(): NavigationMode {
+    if (this.awNavigationMode) {
+      return this.awNavigationMode;
+    }
+    return new ConfigurableNavigationMode(this.navigateBackward, this.navigateForward);
+  }
+
+}
+

```

Comment:
_TODO_: Document all new public APIs

Reply:
NavigationModeDirective` and `ConfigurableNavigationMode` documented in 396faceacc3c509fc7bb179fac1a5a2aa14854b1 and 1888a68652e9ba292590fb367974e678a1c7811c respectively.  Judging by the diff for `index.ts` and `archwizard.module.ts`, there is nothing left to do.